laws-africa/indigo
# Email Notification Context for Annotation Replies

#### What does this PR do?
Adds more context to the email notification about a new comment on a doc.

#### Description of Task to be completed?
When a new reply to a comment on a document is posted, an email is sent out to all the participants of the current thread. This PR adds more context to the email including the thread of comments and a summary of the task linked to the original comment if available.

The contextual data includes:
- [x] initial comment/annotation
- [x] ellipsis if there's more than 3 replies
- [x] last two replies
- [x] task summary

> Task summary is also added to emails sent when a task receives new comment.

#### How should this be manually tested?
- Create a document,
- Create a comment and an optional task linked to the annotation,
- Reply to that comment as other users and check the email sent

#### What are the relevant issues?
closes #731 

#### Screenshots:
Email sent to other users when there are more than 9 replies to a comment:
![image](https://user-images.githubusercontent.com/8082197/65372517-19e3ef80-dc7a-11e9-925c-1e34e7d7da2b.png)

Email sent to other users when there are less than 9 replies to a comment:
![image](https://user-images.githubusercontent.com/8082197/65373377-67198e80-dc85-11e9-9a5b-8dd006e1d0d5.png)

Email sent when a task receives a new comment, it now has the task summary at the bottom:
![image](https://user-images.githubusercontent.com/8082197/65372587-efdefd00-dc7a-11e9-9a84-3937bfb8112c.png)

#### Any background context you want to provide?
The styling on the `annotation_new_reply` email needs some cleanup.
